src/utils/daos_hdlr.c

```diff
@@ -521,6 +521,15 @@ pool_list_containers_hdlr(struct cmd_args_s *ap)
 out_disconnect:
 	/* Pool disconnect  in normal and error flows: preserve rc */
 	rc2 = daos_pool_disconnect(ap->pool, NULL);
+	/* Automatically retry in case of DER_NOMEM.  This is to allow the
+	 * NLT testing to correctly stress-test all code paths and not
+	 * register any memory leaks.
+	 * TODO: Move this retry login into daos_pool_disconnect()
```

Comment:
I agree there is some kind of TODO here. Not sure that it makes sense long term to change product code whether it be daos utility here, or inside the client API implementation in such a way that it is geared toward making a specific test pass. I am not as familiar with all of the layers of reference counting, but had considered whether daos_fini() and in particular the daos_hhash_fini() and d_hhash_destroy() execution is intended to be able to free up client resources for pools that the client is still connected to.

Reply:
My preferred way forward would be to land this, but file a ticket and raise the question of how to do this properly in the PDT meeting.  Then we could update this comment to include the ticket number.

Without this change there are considerable memory leaks, which would restrict the functionality of the fault injection testing.